worker/src/global.rs

```diff
@@ -1,37 +1,75 @@
 use std::ops::Deref;
+use std::time::Duration;
 
+use reqwest::{Body, Client, Error, Response};
+use tokio::time;
 use wasm_bindgen::JsCast;
 use wasm_bindgen_futures::JsFuture;
 
 use crate::{
-    request::Request as WorkerRequest, response::Response as WorkerResponse, AbortSignal, Result,
+     AbortSignal, Result,
 };
 
 /// Construct a Fetch call from a URL string or a Request object. Call its `send` method to execute
 /// the request.
 pub enum Fetch {
     Url(url::Url),
-    Request(WorkerRequest),
+    Request(Body),
 }
 
 impl Fetch {
     /// Execute a Fetch call and receive a Response.
-    pub async fn send(&self) -> Result<WorkerResponse> {
+    pub async fn send(&self) -> Result<Response> {
         match self {
             Fetch::Url(url) => fetch_with_str(url.as_ref(), None).await,
             Fetch::Request(req) => fetch_with_request(req, None).await,
         }
     }
 
     /// Execute a Fetch call and receive a Response.
-    pub async fn send_with_signal(&self, signal: &AbortSignal) -> Result<WorkerResponse> {
+    pub async fn send_with_signal(&self, signal: &AbortSignal) -> Result<Response> {
         match self {
             Fetch::Url(url) => fetch_with_str(url.as_ref(), Some(signal)).await,
             Fetch::Request(req) => fetch_with_request(req, Some(signal)).await,
         }
     }
 }
 
+//#[cfg(feature = "http")]
+async fn fetch_with_str(url: &str, signal: Option<&AbortSignal>) -> Result<Response> {
```

Comment:
Might have to upstream some changes to pass in the abort signal https://github.com/seanmonstar/reqwest/blob/68a3f5803b03393938c39a3794a5a9e5f5264796/src/wasm/client.rs#L221

Reply:
This might be a dumb question, but why does the `reqwest `repo contain `.signal` functionality and the Cargo package doesn't? I don't see it in the docs either when it's clearly in the repository.
https://docs.rs/reqwest/latest/reqwest/struct.Client.html

Reply:
Would it be acceptable to just use `Drop` and provide a handler? I am not sure if we necessarily need direct access to the signal.

Something like:
```rs
// Asynchronous loop to continuously check for cancellation signal
    loop {
        tokio::select! {
            _ = sleep(Duration::from_secs(1)) => {
                println!("Async task running...");
            }
            _ = cancel_receiver.recv() => {
                println!("Cancellation signal received. Dropping task...");
                break; // Break out of the loop when cancellation signal is received
            }
        }
    }```

Reply:
I'm not sure what the tradeoffs are here. Using a JavaScript Abort Signal lets the runtime cancel the request rather which I think might be more efficient, but if there is no way to plumb this through it may be ok to go this route.